spec/custom_attribute_condition_evaluator_spec.rb

```diff
@@ -0,0 +1,271 @@
+    before(:context) do
+      @substring_conditions = {'match' => 'substring', 'name' => 'text', 'type' => 'custom_attribute', 'value' => 'test message!'}
+    end
+
+    it 'should return true if the condition value is a substring of the user-provided value' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('text' => 'This is a test message!')
+      expect(condition_evaluator.evaluate(@substring_conditions)).to be true
+    end
+
+    it 'should return false if the user-provided value is not a substring of the condition value' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('text' => 'Not found!')
+      expect(condition_evaluator.evaluate(@substring_conditions)).to be false
+    end
+
+    it 'should return nil if the user-provided value is not a string' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('text' => 10)
+      expect(condition_evaluator.evaluate(@substring_conditions)).to eq(nil)
+    end
+
+    it 'should return nil if there is no user-provided value' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('text' => {})
+      expect(condition_evaluator.evaluate(@substring_conditions)).to eq(nil)
+    end
+  end
+
+  describe 'greater than match type' do
+    before(:context) do
+      @gt_conditions = {'match' => 'gt', 'name' => 'input_value', 'type' => 'custom_attribute', 'value' => 10}
+    end
+
+    it 'should return true if the user-provided value is greater than the condition value' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => 12)
+      expect(condition_evaluator.evaluate(@gt_conditions)).to be true
+    end
+
+    it 'should return false if the user-provided value is not greater than the condition value' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => 8)
+      expect(condition_evaluator.evaluate(@gt_conditions)).to be false
+    end
+
+    it 'should return nil if the user-provided value is not a number' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => 'test')
+      expect(condition_evaluator.evaluate(@gt_conditions)).to eq(nil)
+    end
+
+    it 'should return nil if there is no user-provided value' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => {})
+      expect(condition_evaluator.evaluate(@gt_conditions)).to eq(nil)
+    end
+
+    it 'should return nil if user-provided value is greater than specified limit' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => 2.0e+53)
+      expect(condition_evaluator.evaluate(@gt_conditions)).to eq(nil)
+    end
+  end
+
+  describe 'less than match type' do
+    before(:context) do
+      @lt_conditions = {'match' => 'lt', 'name' => 'input_value', 'type' => 'custom_attribute', 'value' => 10}
```

Comment:
Both for gt and lt add same testcases where condition value is a  float. See Python PR #146